### PR TITLE
feat: add pdf crafter

### PR DESCRIPTION
**Pull request by theUnkownName**
extract text from pdf

### CODE REVIEW

File: crafters/nlp/PDFExtractor/__init__.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    def post_init(self):
+        import PyPDF2
+        import io
+
+    def craft(self, uri: str, buffer: bytes, *args, **kwargs):
+        import PyPDF2
+        import io
+        text = ""
+        if buffer:
```
Comment (deepankarm) [inline crafters/nlp/PDFExtractor/__init__.py:23]: Instead of using 2 variables `uri` & `buffer`, we can use just 1 and do type checking (`bytes` vs `str`)

Comment (JoanFM) [reply crafters/nlp/PDFExtractor/__init__.py:23]: Driver is passing the info depending on the content I think, similar to what `ImageReader` is having.

File: crafters/nlp/PDFExtractor/__init__.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+class PDFTextExtractor(BaseCrafter):
+    """
+    :class:`PDFTextExtractor` Extracts text from PDF.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def post_init(self):
```
Comment (JoanFM) [inline crafters/nlp/PDFExtractor/__init__.py:15]: Import once, remove from post_init

File: crafters/nlp/PDFExtractor/__init__.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        else:
+            raise ValueError('No value found in "buffer" and "uri"')
+
+
+        pdf_reader = PyPDF2.PdfFileReader(pdf_obj)
+        count = pdf_reader.numPages
+        for i in range(count):
+            page = pdf_reader.getPage(i)
+            text += page.extractText()
+
```
Comment (JoanFM) [inline crafters/nlp/PDFExtractor/__init__.py:36]: too many blank spaces

File: crafters/nlp/PDFExtractor/__init__.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        import PyPDF2
+        import io
+        text = ""
+        if buffer:
+            pdf_obj = io.BytesIO(buffer)
+        elif uri:
+            pdf_obj = open(uri, 'rb')
+        else:
+            raise ValueError('No value found in "buffer" and "uri"')
+
```
Comment (JoanFM) [inline crafters/nlp/PDFExtractor/__init__.py:29]: too many blank spaces

**Review by JoanFM — DISMISSED** [review]

File: crafters/nlp/PDFExtractor/__init__.py (hunk 1 above)
Comment (deepankarm) [reply crafters/nlp/PDFExtractor/__init__.py:23]: Oh! This is coming from the `Document` proto. Got it. Apologies!

Comment (JoanFM) [reply crafters/nlp/PDFExtractor/__init__.py:23]: None apologies needed!

**Review by hanxiao — APPROVED** [review]

**Pull request merged**